### DISCUSSION TITLE
Teensy 4.1 qnethernet 2k limit?

### DISCUSSION THREAD

**Discussion by videojedi** [khoih-prog/EthernetWebServer#49] [Q&A] [answered]
Hi. I've already split my webpage into html, css and script files. Now I've tried adding a small base64 image as a separate PROGMEM header file as well. But I think it's too large and is being truncated by the 2k limit.  

Is there a fix or workaround for this combination of controller and library?

Many thanks for the library. Great work. 

Richard.

**Comment by khoih-prog** [comment]
Normally, large file will be sent by chunks, not the whole file at once.

Check [**ESP32_FS_EthernetWebServer** example](https://github.com/khoih-prog/EthernetWebServer/tree/master/examples/ESP32_FS_EthernetWebServer) where large files, in [**data** folder](https://github.com/khoih-prog/EthernetWebServer/tree/master/examples/ESP32_FS_EthernetWebServer/data) are sent via Ethernet without any issue.

Use that example with ESP32 to test your code, file before moving to **Teensy 4.1 + QNEthernet**

If you have issue with QNEthernet, please post the problem in the [**QNEthernet** issues](https://github.com/ssilverman/QNEthernet/issues)

**Comment by videojedi** [comment]
thanks. I will try an ESP32. 

Should the send_P command work? as I'm getting the following error.

no matching function for call to 'EthernetWebServer::send_P(int, const char [23], String&)'

**Comment by khoih-prog** [comment]
You have to use for Teensy 4.1

```
server.send_P(200, "text/html", temp);
```

instead of

```
server.send_P(200, F("text/html"), temp);
```

**Comment by videojedi** [comment]
Thank you once again. The file is being sent to the browser, but I think I have the content type set incorrectly?

index.h
`<img src="logo" width="50" height="50" alt="logo" />`

logo.h
`const char logo[] PROGMEM =R"rawLiteral(iVBORw0KGgoAAAANSUhEUgAAABQAA ........ )rawLiteral";`

app.ino
` server.on("/logo", handleLogo);`

`void handleLogo(){
 server.send_P(200,"data:image/png;base64,",logo);
}`

The browser doesn't render the image.

**Reply by khoih-prog** [reply]
I'm sorry I can't help you out any further with the basic HTTP(S) usage and not the bug of this library. Try to use other examples and figure out how they can send images.

I suggest you post the issue for help on Teensy or Arduino Forum, if you still have problem.

**Comment by videojedi** [comment, marked as answer]
Ok, I have a solution.

I've added a fetch in the javascript, that pulls the base64 image and replaces the HTML image src with the received text.

There's probably a better way, but this does seem to work.

index.h
`<img src="logo" width="50" height="50" alt="logo" />`

logo.h
`const char logo[] PROGMEM =R"rawLiteral(iVBORw0KGgoAAAANSUhEUgAAABQAA ........ )rawLiteral";`

app.ino
` server.on("/logo", handleLogo);`

`void handleLogo(){
 server.send_P(200,"text/plain",logo);
}`

script.h
```
const url = '/logo'

const getImage = async (url) => {
       return await fetch(url).then(res => res.text())
    }

function getLogo(){
    getImage(url)
      .then(result => {
        document.getElementById('logoImage').src = result
      })
    };

getLogo()
```
